apps/src/blockly/addons/plusMinusBlocks/advancedProcedures.js

```diff
@@ -444,10 +449,7 @@ const procedureDefMutator = {
    * @private
    */
   removeArg_: function (argId) {
-    // Find the argument name by traversing the connection of the arg input.
-    const argLabel = this.getInput(argId)
-      .connection.targetConnection.getSourceBlock()
-      .getFieldValue('VAR');
+    const variable = this.argData_.find(arg => arg.argId === argId);
```

Comment:
Instead of removing blocks based on the field value, we're looking at the actual variable ID, which should theoretically be safer.

Reply:
Could `variable` be `undefined`?

Reply:
It really _shouldn't_ be, but I'll wrap it in a conditional check just to be safe.